File: archive/canvas-api/src/main/java/com/instructure/canvasapi/model/CanvasContextPermission.java
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */

package com.instructure.canvasapi.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class CanvasContextPermission implements Parcelable {

    // Course Permissions
    @SerializedName("create_discussion_topic")
    private boolean canCreateDiscussionTopic;

    // User Permissions
    @SerializedName("can_update_name")
    private boolean canUpdateName;

    @SerializedName("can_update_avatar")
    private boolean canUpdateAvatar;

    @SerializedName("create_announcement")
    private boolean canCreateAnnouncement;

    ///////////////////////////////////////////////////////////////////////////
    // Getters and Setters
    ///////////////////////////////////////////////////////////////////////////

    public boolean canUpdateName() {
        return canUpdateName;
    }

    public void setCanUpdateName(boolean canUpdateName) {
        this.canUpdateName = canUpdateName;
    }

    public boolean canUpdateAvatar() {
        return canUpdateAvatar;
    }

    public void setCanUpdateAvatar(boolean canUpdateAvatar) {
        this.canUpdateAvatar = canUpdateAvatar;
    }

    public boolean canCreateDiscussionTopic(){
        return this.canCreateDiscussionTopic;
    }

    public void setCanCreateDiscussionTopic(boolean canCreateDiscussionTopic){
        this.canCreateDiscussionTopic = canCreateDiscussionTopic;
    }

    public boolean canCreateAnnouncement() {
        return canCreateAnnouncement;
    }

    public void setCanCreateAnnouncement(boolean canCreateAnnouncement) {
        this.canCreateAnnouncement = canCreateAnnouncement;
    }

    ///////////////////////////////////////////////////////////////////////////
    // Constructors
    ///////////////////////////////////////////////////////////////////////////
    public CanvasContextPermission() {}

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeByte(canUpdateName ? (byte) 1 : (byte) 0);
        dest.writeByte(canUpdateAvatar ? (byte) 1 : (byte) 0);
        dest.writeByte(canCreateDiscussionTopic ? (byte) 1 : (byte) 0);
        dest.writeByte(canCreateAnnouncement ? (byte) 1 : (byte) 0);
    }
    private CanvasContextPermission(Parcel in) {
        this.canUpdateName = in.readByte() != 0;
        this.canUpdateAvatar = in.readByte() != 0;
        this.canCreateDiscussionTopic = in.readByte() != 0;
        this.canCreateAnnouncement = in.readByte() != 0;
    }

    public static Creator<CanvasContextPermission> CREATOR = new Creator<CanvasContextPermission>() {
        public CanvasContextPermission createFromParcel(Parcel source) {
            return new CanvasContextPermission(source);
        }

        public CanvasContextPermission[] newArray(int size) {
            return new CanvasContextPermission[size];
        }
    };

}

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/models/CanvasContextPermission.kt
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */
package com.instructure.canvasapi2.models

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
data class CanvasContextPermission(
        @SerializedName(CREATE_DISCUSSION_TOPIC)
        val canCreateDiscussionTopic: Boolean = false,
        @SerializedName(MANAGE_GRADES)
        val manage_grades: Boolean = false,
        @SerializedName(SEND_MESSAGES)
        val send_messages: Boolean = false,
        @SerializedName(SEND_MESSAGES_ALL)
        val send_messages_all: Boolean = false,
        @SerializedName(VIEW_ALL_GRADES)
        val view_all_grades: Boolean = false,
        @SerializedName(VIEW_ANALYTICS)
        val view_analytics: Boolean = false,
        @SerializedName(BECOME_USER)
        val become_user: Boolean = false,
        @SerializedName(CAN_UPDATE_NAME)
        val canUpdateName: Boolean = false,
        @SerializedName(CAN_UPDATE_AVATAR)
        val canUpdateAvatar: Boolean = false,
        @SerializedName(CREATE_ANNOUNCEMENT)
        val canCreateAnnouncement: Boolean = false,
        @SerializedName(STUDENT_VIEW)
        val canUseStudentView: Boolean = false,
        @SerializedName(MANAGE_CONTENT)
        val canManageContent: Boolean = false,
        @SerializedName(MANAGE_COURSE_CONTENT_EDIT)
        val canEditCourseContent: Boolean = false,
        @SerializedName(MANAGE_CALENDAR)
        val manageCalendar: Boolean = false,
        @SerializedName(LIMIT_PARENT_APP_WEB_ACCESS)
        val limitParentAppWebAccess: Boolean = false
) : Parcelable {
    companion object {
        const val BECOME_USER = "become_user"
        const val CAN_UPDATE_AVATAR = "can_update_avatar"
        const val CAN_UPDATE_NAME = "can_update_name"
        const val CREATE_ANNOUNCEMENT = "create_announcement"
        const val CREATE_DISCUSSION_TOPIC = "create_discussion_topic"
        const val DELETE = "delete"
        const val MANAGE_GRADES = "manage_grades"
        const val SEND_MESSAGES = "send_messages"
        const val SEND_MESSAGES_ALL = "send_messages_all"
        const val VIEW_ALL_GRADES = "view_all_grades"
        const val VIEW_ANALYTICS = "view_analytics"
        const val STUDENT_VIEW = "use_student_view"
        const val MANAGE_CONTENT = "manage_content"
        const val MANAGE_COURSE_CONTENT_EDIT = "manage_course_content_edit"
        const val MANAGE_CALENDAR = "manage_calendar"
        const val LIMIT_PARENT_APP_WEB_ACCESS = "limit_parent_app_web_access"
    }
}
